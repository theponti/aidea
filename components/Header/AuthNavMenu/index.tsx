"use client";

import * as DropdownMenu from "@radix-ui/react-dropdown-menu";
import { Menu } from "lucide-react";
import { Session } from "next-auth";
import { signIn, signOut } from "next-auth/react";
import Link from "next/link";

import Avatar from "components/Avatar";
import { ACCOUNT_PATH } from "lib/routes";

import styles from "./AuthNavMenu.module.css";

export default function AuthMenu({ session }: { session: Session | null }) {
  const pictureUrl = session?.user?.image as string;

  if (!session) {
    return (
      <button onClick={() => signIn("google")} data-testid="signInButton">
        Sign In
      </button>
    );
  }

  return (
    <div
      data-testid="AuthenticatedMenu"
      className="flex border-1 border-solid border-slateblue rounded-full p-4 hover:shadow-md hover:shadow-slateblue"
    >
      <Avatar src={pictureUrl} alt={session?.user?.name || ""} />
      <DropdownMenu.Root>
        <DropdownMenu.Trigger asChild>
          <button className={styles.IconButton} aria-label="account menu">
            <Menu />
          </button>
        </DropdownMenu.Trigger>
        <DropdownMenu.Portal>
          <DropdownMenu.Content
            className={styles.DropdownMenuContent}
            align="end"
            sideOffset={5}
          >
            <DropdownMenu.Item className={styles.DropdownMenuItem}>
              <Link href={ACCOUNT_PATH}>Account</Link>
            </DropdownMenu.Item>
            <DropdownMenu.Separator className={styles.DropdownMenuSeparator} />
            <DropdownMenu.Item
              className={styles.DropdownMenuItem}
              onClick={() => signOut()}
            >
              Logout
            </DropdownMenu.Item>
          </DropdownMenu.Content>
        </DropdownMenu.Portal>
      </DropdownMenu.Root>
    </div>
  );
}
