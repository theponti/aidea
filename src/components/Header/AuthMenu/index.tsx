import { signIn, useSession } from "next-auth/react";
import Link from "next/link";
import { useCallback } from "react";

interface AuthMenuButtonProps {
  status: string;
}
function AuthMenuButton({ status }: AuthMenuButtonProps) {
  const onSignInClick = useCallback(() => {
    signIn("google");
  }, []);

  if (status === "unauthenticated") {
    return (
      <button onClick={onSignInClick} data-testid="signInButton">
        Sign In
      </button>
    );
  }

  return (
    <Link data-testid="accountLink" href="/account">
      My Account
    </Link>
  );
}

function AuthMenu() {
  const { status } = useSession();

  if (status === "loading") {
    return null;
  }

  return (
    <div className="flex-none gap-2">
      <ul className="menu menu-horizontal p-0">
        <li>
          <AuthMenuButton status={status} />
        </li>
      </ul>
    </div>
  );
}

export default AuthMenu;
