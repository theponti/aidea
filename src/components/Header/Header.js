import React from 'react'
import t from 'prop-types'
import styles from './Header.module.scss'

export const Header = ({ children }) => {
  return (
    <div className={styles.container}>
      <h1>{children}</h1>
    </div>
  )
}

Header.propTypes = {
  children: t.node
}

export default Header
