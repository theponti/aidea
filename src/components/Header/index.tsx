import { signIn, useSession } from "next-auth/react";
import Link from "next/link";

function Header() {
  const { status } = useSession();
  return (
    <nav className="navbar bg-base-100 sticky top-0 left-0 px-0 drop-shadow-sm">
      <div className="container mx-auto max-w-[1200px] px-4">
        <div className="flex-1 text-primary">
          <span className="font-semibold text-2xl">
            <Link href="/">ðŸ§  Aidea</Link>
          </span>
        </div>
        {status !== "loading" && (
          <div className="flex-none gap-2">
            <ul className="menu menu-horizontal p-0">
              <li>
                {status === "unauthenticated" ? (
                  <button onClick={() => signIn("google")}>Sign In</button>
                ) : (
                  <Link data-testid="accountLink" href="/account">
                    My Account
                  </Link>
                )}
              </li>
            </ul>
          </div>
        )}
      </div>
    </nav>
  );
}

export default Header;
