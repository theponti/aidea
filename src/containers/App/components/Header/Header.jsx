import { useAuth0 } from "@auth0/auth0-react";
import styled from "@emotion/styled";
import PropTypes from "prop-types";
import { Fragment } from "react";
import { Link } from "react-router-dom";
import Brain from "src/components/Brain";
import Button from "src/components/Button";
import constants from "src/constants";
import { ACCOUNT_PATH } from "src/constants/routes";
import styles from "./Header.module.scss";

function MenuItem({ children }) {
  return <div className={styles.menuItem}>{children}</div>;
}

MenuItem.propTypes = {
  children: PropTypes.node,
};

const GetStartedButton = styled(Button)`
  padding: 12px 16px;
  background-color: black;
`;

function Header() {
  const { isAuthenticated, loginWithPopup, logout } = useAuth0();

  return (
    <header className={styles.header}>
      <div className={styles.rightNav}>
        <Link to="ideas" className={styles.appName}>
          <Brain size={40} />
          {constants.APP_NAME}
        </Link>
      </div>
      <div className={styles.leftNav}>
        {isAuthenticated ? (
          <Fragment>
            <Link to={ACCOUNT_PATH} className={styles.menuItem}>
              Account
            </Link>
            <MenuItem onClick={logout}>Log Out</MenuItem>
          </Fragment>
        ) : (
          <Fragment>
            <MenuItem onClick={loginWithPopup}>Log In</MenuItem>
            <GetStartedButton onClick={loginWithPopup}>
              Get Started
            </GetStartedButton>
          </Fragment>
        )}
      </div>
    </header>
  );
}

export default Header;
