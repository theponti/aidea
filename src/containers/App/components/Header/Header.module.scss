.header {
    display: flex;
    align-content: center;
    justify-content: space-between;
    box-shadow: 0 0.01em 0.3rem 0 #797979;
    margin: 0 0 2.5rem;
    padding: 0.75rem 1rem;
}

.appName {
    display: flex;
    align-items: center;
    font-size: 1.5rem;

    img {
        height: 2.5rem;
        margin-right: 0.4rem;
    }
}

.leftNav {
    display: flex;
    align-items: center;
}

.menuItem {
    @apply font-medium;
    @apply tracking-wide;
    display: inline-block;
    margin: 0 1rem;
    opacity: 0.6;
}
