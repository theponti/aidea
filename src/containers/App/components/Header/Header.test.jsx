import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { describe, expect, it, vi } from "vitest";
import Header from "./Header";

describe("<Header/>", () => {
  it("should render component", () => {
    const { container } = render(
      <BrowserRouter>
        <Header login={vi.fn()} />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
