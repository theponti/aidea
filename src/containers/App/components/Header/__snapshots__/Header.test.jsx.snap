// Vitest Snapshot v1

exports[`<Header/> > should render authenticated header 1`] = `
<div>
  <header
    class="_header_1qsem_1"
  >
    <div>
      <a
        class="_appName_1qsem_9"
        href="/ideas"
      >
        <svg
          height="40"
          id="brain emoji"
          viewBox="0 0 72 72"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="color"
          >
            <line
              fill="#ea5a47"
              stroke="none"
              x1="16"
              x2="25.875"
              y1="35"
              y2="35"
            />
            <path
              d="M25.875,35c3.4167,0,4.75-2.9167,4.75-6.1667"
              fill="#ea5a47"
              stroke="none"
            />
            <line
              fill="#ea5a47"
              stroke="none"
              x1="16"
              x2="19.75"
              y1="45.5"
              y2="45.5"
            />
            <line
              fill="#ea5a47"
              stroke="none"
              x1="35.2997"
              x2="35.2997"
              y1="45.5"
              y2="39.4375"
            />
            <path
              d="M35.2997,39.4375c0-2.2455,1.2003-5.25,5.0756-5.25"
              fill="#ea5a47"
              stroke="none"
            />
            <line
              fill="#ea5a47"
              stroke="none"
              x1="58.4583"
              x2="49.875"
              y1="34.5417"
              y2="34.5417"
            />
            <path
              d="M40.375,45.5c0,0,0.1642,10.5-7.9583,10.5H16c-7.2083,0-7.6667-10.5,0-10.5C8.7917,45.5,8.3333,35,16,35 c-8.1667-7.25,1.875-13.5625,6.5-8.6875c-4.799-6.625,5.375-10.3125,8.875-5C30.9627,16,36.2222,15.0417,41.1003,19.024 l-0.4336,0.351c2.375-2.375,10.1776-1.0833,10.1776,5.3141l-2.2192,1.6234c6.8646-6.8646,14.2083,4.1875,8.5,8.2292h1.3333 c4.75,0,5.2917,10.9583-1.3333,10.9583H29.2917"
              fill="#FFA7C0"
              stroke="none"
            />
            <path
              d="M40.6667,19.375c-2.4167,2.4167-0.2822,3.1782-4.2707,7.1667"
              fill="#ea5a47"
              stroke="none"
            />
          </g>
          <g
            id="hair"
          />
          <g
            id="skin"
          />
          <g
            id="skin-shadow"
          />
          <g
            id="line"
          >
            <line
              fill="none"
              stroke="#000000"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-miterlimit="10"
              stroke-width="2"
              x1="16"
              x2="25.875"
              y1="35"
              y2="35"
            />
            <path
              d="M25.875,35c3.4167,0,4.75-2.9167,4.75-6.1667"
              fill="none"
              stroke="#000000"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-miterlimit="10"
              stroke-width="2"
            />
            <line
              fill="none"
              stroke="#000000"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-miterlimit="10"
              stroke-width="2"
              x1="16"
              x2="19.75"
              y1="45.5"
              y2="45.5"
            />
            <line
              fill="none"
              stroke="#000000"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-miterlimit="10"
              stroke-width="2"
              x1="35.2997"
              x2="35.2997"
              y1="45.5"
              y2="39.4375"
            />
            <path
              d="M35.2997,39.4375c0-2.2455,1.2003-5.25,5.0756-5.25"
              fill="none"
              stroke="#000000"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-miterlimit="10"
              stroke-width="2"
            />
            <line
              fill="none"
              stroke="#000000"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-miterlimit="10"
              stroke-width="2"
              x1="58.4583"
              x2="49.875"
              y1="34.5417"
              y2="34.5417"
            />
            <path
              d="M40.375,45.5c0,0,0.1642,10.5-7.9583,10.5H16c-7.2083,0-7.6667-10.5,0-10.5C8.7917,45.5,8.3333,35,16,35 c-8.1667-7.25,1.875-13.5625,6.5-8.6875c-4.799-6.625,5.375-10.3125,8.875-5C30.9627,16,36.2222,15.0417,41.1003,19.024 l-0.4336,0.351c2.375-2.375,10.1776-1.0833,10.1776,5.3141l-2.2192,1.6234c6.8646-6.8646,14.2083,4.1875,8.5,8.2292h1.3333 c4.75,0,5.2917,10.9583-1.3333,10.9583H29.2917"
              fill="none"
              stroke="#000000"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-miterlimit="10"
              stroke-width="2"
            />
            <path
              d="M40.6667,19.375c-2.4167,2.4167-0.2822,3.1782-4.2707,7.1667"
              fill="none"
              stroke="#000000"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-miterlimit="10"
              stroke-width="2"
            />
          </g>
        </svg>
        Aidea
      </a>
    </div>
    <div
      class="_leftNav_1qsem_22"
    >
      <a
        class="_menuItem_1qsem_27"
        href="/account"
      >
        Account
      </a>
      <div
        class="_menuItem_1qsem_27"
      >
        Log Out
      </div>
    </div>
  </header>
</div>
`;

exports[`<Header/> > should render unauthenticated header 1`] = `
<div>
  <header
    class="_header_1qsem_1"
  >
    <div>
      <a
        class="_appName_1qsem_9"
        href="/ideas"
      >
        <svg
          height="40"
          id="brain emoji"
          viewBox="0 0 72 72"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="color"
          >
            <line
              fill="#ea5a47"
              stroke="none"
              x1="16"
              x2="25.875"
              y1="35"
              y2="35"
            />
            <path
              d="M25.875,35c3.4167,0,4.75-2.9167,4.75-6.1667"
              fill="#ea5a47"
              stroke="none"
            />
            <line
              fill="#ea5a47"
              stroke="none"
              x1="16"
              x2="19.75"
              y1="45.5"
              y2="45.5"
            />
            <line
              fill="#ea5a47"
              stroke="none"
              x1="35.2997"
              x2="35.2997"
              y1="45.5"
              y2="39.4375"
            />
            <path
              d="M35.2997,39.4375c0-2.2455,1.2003-5.25,5.0756-5.25"
              fill="#ea5a47"
              stroke="none"
            />
            <line
              fill="#ea5a47"
              stroke="none"
              x1="58.4583"
              x2="49.875"
              y1="34.5417"
              y2="34.5417"
            />
            <path
              d="M40.375,45.5c0,0,0.1642,10.5-7.9583,10.5H16c-7.2083,0-7.6667-10.5,0-10.5C8.7917,45.5,8.3333,35,16,35 c-8.1667-7.25,1.875-13.5625,6.5-8.6875c-4.799-6.625,5.375-10.3125,8.875-5C30.9627,16,36.2222,15.0417,41.1003,19.024 l-0.4336,0.351c2.375-2.375,10.1776-1.0833,10.1776,5.3141l-2.2192,1.6234c6.8646-6.8646,14.2083,4.1875,8.5,8.2292h1.3333 c4.75,0,5.2917,10.9583-1.3333,10.9583H29.2917"
              fill="#FFA7C0"
              stroke="none"
            />
            <path
              d="M40.6667,19.375c-2.4167,2.4167-0.2822,3.1782-4.2707,7.1667"
              fill="#ea5a47"
              stroke="none"
            />
          </g>
          <g
            id="hair"
          />
          <g
            id="skin"
          />
          <g
            id="skin-shadow"
          />
          <g
            id="line"
          >
            <line
              fill="none"
              stroke="#000000"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-miterlimit="10"
              stroke-width="2"
              x1="16"
              x2="25.875"
              y1="35"
              y2="35"
            />
            <path
              d="M25.875,35c3.4167,0,4.75-2.9167,4.75-6.1667"
              fill="none"
              stroke="#000000"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-miterlimit="10"
              stroke-width="2"
            />
            <line
              fill="none"
              stroke="#000000"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-miterlimit="10"
              stroke-width="2"
              x1="16"
              x2="19.75"
              y1="45.5"
              y2="45.5"
            />
            <line
              fill="none"
              stroke="#000000"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-miterlimit="10"
              stroke-width="2"
              x1="35.2997"
              x2="35.2997"
              y1="45.5"
              y2="39.4375"
            />
            <path
              d="M35.2997,39.4375c0-2.2455,1.2003-5.25,5.0756-5.25"
              fill="none"
              stroke="#000000"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-miterlimit="10"
              stroke-width="2"
            />
            <line
              fill="none"
              stroke="#000000"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-miterlimit="10"
              stroke-width="2"
              x1="58.4583"
              x2="49.875"
              y1="34.5417"
              y2="34.5417"
            />
            <path
              d="M40.375,45.5c0,0,0.1642,10.5-7.9583,10.5H16c-7.2083,0-7.6667-10.5,0-10.5C8.7917,45.5,8.3333,35,16,35 c-8.1667-7.25,1.875-13.5625,6.5-8.6875c-4.799-6.625,5.375-10.3125,8.875-5C30.9627,16,36.2222,15.0417,41.1003,19.024 l-0.4336,0.351c2.375-2.375,10.1776-1.0833,10.1776,5.3141l-2.2192,1.6234c6.8646-6.8646,14.2083,4.1875,8.5,8.2292h1.3333 c4.75,0,5.2917,10.9583-1.3333,10.9583H29.2917"
              fill="none"
              stroke="#000000"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-miterlimit="10"
              stroke-width="2"
            />
            <path
              d="M40.6667,19.375c-2.4167,2.4167-0.2822,3.1782-4.2707,7.1667"
              fill="none"
              stroke="#000000"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-miterlimit="10"
              stroke-width="2"
            />
          </g>
        </svg>
        Aidea
      </a>
    </div>
    <div
      class="_leftNav_1qsem_22"
    >
      <div
        class="_menuItem_1qsem_27"
      >
        Log In
      </div>
      <button
        class="_container_1pont_1 css-rduvsv"
        data-testid="button"
      >
        Get Started
      </button>
    </div>
  </header>
</div>
`;
